azkaban-execserver/src/main/java/azkaban/execapp/ServerStatisticsServlet.java

```diff
@@ -91,30 +91,70 @@ protected void fillRemainingMemoryPercent(ExecutorInfo stats){
       }
 
       long totalMemory = 0;
-      long  freeMemory = 0;
+      long totalFreeMemory = 0;
+      long parsedResult = 0;
 
       // process the output from bash call.
       // we expect the result from the bash call to be something like following -
       // MemTotal:       65894264 kB
-      // MemFree:        61104076 kB
-      if (output.size() == 2) {
+      // MemFree:        57753844 kB
+      // Buffers:          305552 kB
+      // Cached:          3802432 kB
+      // SwapCached:            0 kB
+      // Note : total free memory = freeMemory + cached + buffers + swapCached
+      if (output.size() == 5) {
         for (String result : output){
           // find the total memory and value the variable.
           if (result.contains("MemTotal") && result.split("\\s+").length > 2){
             try {
               totalMemory = Long.parseLong(result.split("\\s+")[1]);
-              logger.info("Total memory : " + totalMemory);
+              logger.debug("Total memory : " + totalMemory);
             }catch(NumberFormatException e){
               logger.error("yielding 0 for total memory as output is invalid -" + result);
             }
           }
-          // find the free memory and value the variable.
+
+          // find the free memory.
           if (result.contains("MemFree") && result.split("\\s+").length > 2){
             try {
-              freeMemory = Long.parseLong(result.split("\\s+")[1]);
-              logger.info("Free memory : " + freeMemory);
+              parsedResult = Long.parseLong(result.split("\\s+")[1]);
+              logger.debug("Free memory : " + parsedResult);
+              totalFreeMemory += parsedResult;
             }catch(NumberFormatException e){
-              logger.error("yielding 0 for total memory as output is invalid -" + result);
+              logger.error("yielding 0 for free memory as output is invalid -" + result);
+            }
+          }
+
+          // find the Buffers.
+          if (result.contains("Buffers") && result.split("\\s+").length > 2){
+            try {
+              parsedResult = Long.parseLong(result.split("\\s+")[1]);
```

Comment:
Is it possible to re-write this code snippet into a function as it is used multiple times